hkmc2/shared/src/main/scala/hkmc2/semantics/ucs/Normalization.scala

```diff
@@ -154,8 +183,9 @@ class Normalization(elaborator: Elaborator)(using raise: Raise, ctx: Ctx):
   
   /**
     * Specialize `split` with the assumption that `scrutinee` matches `pattern`.
-    * If `matchOrNot` is `true`, the function _keeps_ branches that agree on
-    * `scrutinee` matches `pattern`. Otherwise, the function _removes_ branches
+    * If `mode` is `+`, the function _keeps_ branches that agree on
+    * `scrutinee` matches `pattern` and simplifies the record patterns it sees if the fields were already matched.
+    * Otherwise (if `mode is `-`), the function _removes_ branches
```

Comment:
```suggestion
    * `scrutinee` matching `pattern` and simplifies the record patterns it sees if the fields were already matched.
    * Otherwise (if `mode` is `-`), the function _removes_ branches
```